src/mono/wasm/debugger/DebuggerTestSuite/TestHarnessStartup.cs

```diff
@@ -178,7 +178,7 @@ public void Configure(IApplicationBuilder app, IOptionsMonitor<TestHarnessOption
                         var client = new HttpClient();
                         var psi = new ProcessStartInfo();
 
-                        psi.Arguments = $"--headless --disable-gpu --lang=en-US --incognito --remote-debugging-port={devToolsUrl.Port} http://{TestHarnessProxy.Endpoint.Authority}/{options.PagePath}";
+                        psi.Arguments = $"--headless --disable-gpu --lang=en-US --incognito --no-sandbox --remote-debugging-port={devToolsUrl.Port} http://{TestHarnessProxy.Endpoint.Authority}/{options.PagePath}";
```

Comment:
Why is `--no-sandbox` needed?

Reply:
Without it we are getting an exception:

```
The active test run was aborted. Reason: Test host process crashed : Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'input')
   at System.Text.RegularExpressions.ThrowHelper.ThrowArgumentNullException(ExceptionArgument arg)
   at System.Text.RegularExpressions.Regex.Match(String input)
   at Microsoft.WebAssembly.Diagnostics.TestHarnessStartup.<>c__DisplayClass14_0.<LaunchAndServe>b__0(Object sender, DataReceivedEventArgs e) in /workspaces/runtime/src/mono/wasm/debugger/DebuggerTestSuite/TestHarnessStartup.cs:line 102
   at System.Diagnostics.Process.ErrorReadNotifyUser(String data)
   at System.Diagnostics.AsyncStreamReader.FlushMessageQueue(Boolean rethrowInNewThread)
--- End of stack trace from previous location ---
   at System.Diagnostics.AsyncStreamReader.<>c.<FlushMessageQueue>b__18_0(Object edi)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()


Test Run Aborted with error System.Exception: One or more errors occurred.
 ---> System.Exception: Unable to read beyond the end of the stream.
   at System.IO.BinaryReader.Read7BitEncodedInt()
   at System.IO.BinaryReader.ReadString()
   at Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.LengthPrefixCommunicationChannel.NotifyDataAvailable()
   at Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.TcpClientExtensions.MessageLoopAsync(TcpClient client, ICommunicationChannel channel, Action`1 errorHandler, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---.
```

And adding --no-sandbox eliminates it. The idea was originally @thaystg's so I don't know how she found this solution but it works.

Reply:
Looks like it is this code passing `null` into Regex.Match:

https://github.com/dotnet/runtime/blob/a73d44f1009af7954df75e9923c64a3f10012004/src/mono/wasm/debugger/DebuggerTestSuite/TestHarnessStartup.cs#L94-L107

Reply:
@ilonatommy when you run without `--no-sandbox`, can you share the `browser-stderr:` lines?

Reply:
Sure, it looks like this:

```
[dbg] browser-stderr: Failed to move to new namespace: PID namespaces supported, Network namespace supported, but failed: errno = Operation not permitted
[dbg] browser-stderr: [0204/160213.844976:FATAL:zygote_host_impl_linux.cc(189)] Check failed: ReceiveFixedMessage(fds[0], kZygoteBootMessage, sizeof(kZygoteBootMessage), &boot_pid). 
[dbg] browser-stderr: #0 0x557dd62db0a9 (/usr/lib/chromium/chromium+0x536a0a8)
[dbg] browser-stderr: #1 0x557dd623a393 (/usr/lib/chromium/chromium+0x52c9392)
[dbg] browser-stderr: #2 0x557dd624c806 (/usr/lib/chromium/chromium+0x52db805)
[dbg] browser-stderr: #3 0x557dd75ede47 (/usr/lib/chromium/chromium+0x667ce46)
[dbg] browser-stderr: #4 0x557dd5e2dc26 (/usr/lib/chromium/chromium+0x4ebcc25)
[dbg] browser-stderr: #5 0x557dd75ecc31 (/usr/lib/chromium/chromium+0x667bc30)
[dbg] browser-stderr: #6 0x557dd75f3645 (/usr/lib/chromium/chromium+0x6682644)
[dbg] browser-stderr: #7 0x557dd5e2d3d4 (/usr/lib/chromium/chromium+0x4ebc3d3)
[dbg] browser-stderr: #8 0x557dd5e4ad10 (/usr/lib/chromium/chromium+0x4ed9d0f)
[dbg] browser-stderr: #9 0x557dd5e2bd71 (/usr/lib/chromium/chromium+0x4ebad70)
[dbg] browser-stderr: #10 0x557dd5e4a02d (/usr/lib/chromium/chromium+0x4ed902c)
[dbg] browser-stderr: #11 0x557dd5e49d78 (/usr/lib/chromium/chromium+0x4ed8d77)
[dbg] browser-stderr: #12 0x557dd39e3cd7 ChromeMain
[dbg] browser-stderr: #13 0x7f93b3a190b3 __libc_start_main
[dbg] browser-stderr: #14 0x557dd39e3b6e _start
[dbg] browser-stderr: 
[dbg] browser-stderr: Received signal 6
[dbg] browser-stderr: #0 0x557dd62db0a9 (/usr/lib/chromium/chromium+0x536a0a8)
[dbg] browser-stderr: #1 0x557dd623a393 (/usr/lib/chromium/chromium+0x52c9392)
[dbg] browser-stderr: #2 0x557dd62dac90 (/usr/lib/chromium/chromium+0x5369c8f)
[dbg] browser-stderr: #3 0x7f93b82543c0 (/usr/lib/x86_64-linux-gnu/libpthread-2.31.so+0x153bf)
[dbg] browser-stderr: #4 0x7f93b3a3818b gsignal
[dbg] browser-stderr: #5 0x7f93b3a17859 abort
[dbg] browser-stderr: #6 0x557dd62d9c05 (/usr/lib/chromium/chromium+0x5368c04)
[dbg] browser-stderr: #7 0x557dd624cc50 (/usr/lib/chromium/chromium+0x52dbc4f)
[dbg] browser-stderr: #8 0x557dd75ede47 (/usr/lib/chromium/chromium+0x667ce46)
[dbg] browser-stderr: #9 0x557dd5e2dc26 (/usr/lib/chromium/chromium+0x4ebcc25)
[dbg] browser-stderr: #10 0x557dd75ecc31 (/usr/lib/chromium/chromium+0x667bc30)
[dbg] browser-stderr: #11 0x557dd75f3645 (/usr/lib/chromium/chromium+0x6682644)
[dbg] browser-stderr: #12 0x557dd5e2d3d4 (/usr/lib/chromium/chromium+0x4ebc3d3)
[dbg] browser-stderr: #13 0x557dd5e4ad10 (/usr/lib/chromium/chromium+0x4ed9d0f)
[dbg] browser-stderr: #14 0x557dd5e2bd71 (/usr/lib/chromium/chromium+0x4ebad70)
[dbg] browser-stderr: #15 0x557dd5e4a02d (/usr/lib/chromium/chromium+0x4ed902c)
[dbg] browser-stderr: #16 0x557dd5e49d78 (/usr/lib/chromium/chromium+0x4ed8d77)
[dbg] browser-stderr: #17 0x557dd39e3cd7 ChromeMain
[dbg] browser-stderr: #18 0x7f93b3a190b3 __libc_start_main
[dbg] browser-stderr: #19 0x557dd39e3b6e _start
[dbg] browser-stderr:   r8: 0000000000000000  r9: 00007ffcb1e84b60 r10: 0000000000000008 r11: 0000000000000246
[dbg] browser-stderr:  r12: 00000000000000a0 r13: 00007ffcb1e85f38 r14: 00007f93b3bde780 r15: 00007ffcb1e85f30
[dbg] browser-stderr:   di: 0000000000000002  si: 00007ffcb1e84b60  bp: 00007ffcb1e84db0  bx: 00007f93ad253140
[dbg] browser-stderr:   dx: 0000000000000000  ax: 0000000000000000  cx: 00007f93b3a3818b  sp: 00007ffcb1e84b60
[dbg] browser-stderr:   ip: 00007f93b3a3818b efl: 0000000000000246 cgf: 002b000000000033 erf: 0000000000000000
[dbg] browser-stderr:  trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[dbg] browser-stderr: [end of stack trace]
[dbg] browser-stderr: Calling _exit(1). Core file will not be generated.
[dbg] browser-stderr: [0100/000000.912493:ERROR:zygote_linux.cc(653)] write: Broken pipe (32)
[dbg] browser-stderr: 
The active test run was aborted. Reason: Test host process crashed : Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'input')
```

Reply:
I'm not sure if we should use `--no-sandbox`. @kg @lewing thoughts?

Also, add a `string.IsNullOrEmpty` check before line 102.

Reply:
```--no-sandbox``` is different enough from regular end user configurations that we should avoid it whenever possible. historically it has had its own set of bugs and has also avoided certain bugs/perf issues that regular chrome configurations have

Reply:
So we should add --no-sandbox only when we are in codespaces/docker?
Something like @lambdageek did here: https://github.com/dotnet/runtime/blob/689e84bb9be6775d9a401f025494b394ef4241cf/src/mono/sample/wasm/wasm.mk#L14

Reply:
Looks like the issue is that you can't sandbox inside a container without some elevated privs so we need to see if there are other options that will work inside a codespaces.